File: Flora backend/Flora.Services/Database/DecorationSelection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Flora.Services.Database
{
    public class DecorationSelection
    {
        public int Id { get; set; }
        public int DecorationRequestId { get; set; }
        public DecorationRequest DecorationRequest { get; set; } = null!;
        public int DecorationSuggestionId { get; set; }
        public DecorationSuggestion DecorationSuggestion { get; set; } = null!;
        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public string? Comments { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public string Status { get; set; } = "Selected";
    }
}

File: Flora frontend/flora_mobile_app/lib/models/decoration_selection.dart
class DecorationSelection {
  final int id;
  final int decorationRequestId;
  final int decorationSuggestionId;
  final int userId;
  final String? comments;
  final DateTime createdAt;
  final String status;

  DecorationSelection({
    required this.id,
    required this.decorationRequestId,
    required this.decorationSuggestionId,
    required this.userId,
    this.comments,
    required this.createdAt,
    required this.status,
  });

  factory DecorationSelection.fromJson(Map<String, dynamic> json) {
    return DecorationSelection(
      id: json['id'],
      decorationRequestId: json['decorationRequestId'],
      decorationSuggestionId: json['decorationSuggestionId'],
      userId: json['userId'],
      comments: json['comments'],
      createdAt: DateTime.parse(json['createdAt']),
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'decorationRequestId': decorationRequestId,
      'decorationSuggestionId': decorationSuggestionId,
      'userId': userId,
      'comments': comments,
    };
  }
}
